rest/config.go

```diff
@@ -331,7 +334,122 @@ func (dbConfig *DbConfig) setup(name string) error {
+				Message: "error loading sync function",
+				Path:    *dbConfig.Sync,
+				Err:     err,
+			}
+		}
+		dbConfig.Sync = &sync
+	}
+
+	// Load Import Filter Function.
+	if dbConfig.ImportFilter != nil {
+		importFilter, err := loadJavaScript(*dbConfig.ImportFilter)
+		if err != nil {
+			return JavaScriptLoadError{
+				Message: "error loading import filter function",
+				Path:    *dbConfig.ImportFilter,
+				Err:     err,
+			}
+		}
+		dbConfig.ImportFilter = &importFilter
+	}
+
+	// Load Conflict Resolution Function.
+	for _, rc := range dbConfig.Replications {
+		if rc.ConflictResolutionFn != "" {
+			conflictResolutionFn, err := loadJavaScript(rc.ConflictResolutionFn)
+			if err != nil {
+				return JavaScriptLoadError{
+					Message: "error loading conflict resolution function",
+					Path:    rc.ConflictResolutionFn,
+					Err:     err,
+				}
+			}
+			rc.ConflictResolutionFn = conflictResolutionFn
+		}
+	}
+
+	return nil
+}
+
+// loadJavaScript loads the JavaScript source from an external file or and HTTP/HTTPS endpoint.
+// If the specified path does not qualify for a valid file or an URI, it returns the input path
+// as-is with the assumption that it is an inline JavaScript source. Returns error if there is
+// any failure in reading the JavaScript file or URI.
+func loadJavaScript(path string) (js string, err error) {
+	rc, err := readCloser(path)
+	if err != nil {
+		switch err.(type) {
+		case PathNotFoundError:
+			// If rc is nil and readCloser returns no error, treat the
+			// the given path as an inline JavaScript and return it as-is.
+			return path, nil
+		default:
+			return "", err
+		}
+	}
+	defer func() { _ = rc.Close() }()
+	src, err := ioutil.ReadAll(rc)
+	if err != nil {
+		return "", err
+	}
+	return string(src), nil
+}
+
+// JavaScriptLoadError is returned if there is any failure in loading JavaScript
+// source from an external file or URL (HTTP/HTTPS endpoint).
+type JavaScriptLoadError struct {
+	Message string // A short description about the error.
+	Path    string // Path of the JavaScript source.
+	Err     error  // Underlying error.
+}
+
+// Error returns string representation of the JavaScriptLoadError.
+func (e JavaScriptLoadError) Error() string {
+	return fmt.Sprintf("Error loading JavaScript, Message: %s, Path: %s, Err: %v", e.Message, e.Path, e.Err)
+}
+
+// PathNotFoundError is returned when the specified path or URL (HTTP/HTTPS endpoint)
+// doesn't exist while constructing the ReadCloser.
+type PathNotFoundError struct {
+	Path string
+}
+
+// Error returns string representation of the PathNotFoundError.
+func (e PathNotFoundError) Error() string {
+	return fmt.Sprintf("specified path %q doesn't look like a file or URL", e.Path)
+}
+
+// readCloser creates a ReadCloser from the given path. The path must be either a valid file
+// or an HTTP/HTTPS endpoint. Returns an error if there is any failure in building ReadCloser.
+func readCloser(path string) (rc io.ReadCloser, err error) {
```

Comment:
readCloser is too generic of a name for this function - include something to indicate it's attempting to parse and retrieve from a path.

Reply:
Sure Adam, readCloser has been renamed to readFromPath.